.github/workflows/release_github.yml

```diff
@@ -39,119 +85,120 @@ jobs:
+  
+      - name: Install bump-my-version
+        shell: bash
+        run: pip install "bump-my-version==0.28.1"
+        
+      - name: Pass Inputs to Shell
         id: bump
+        shell: bash
         run: |
-          echo "-- install bump-my-version"
-          python3 -m pip install bump-my-version
-          echo "-- bump the version"
-          bump-my-version bump ${{ github.event.inputs.version_level }} --commit --tag
-          echo "-- push bumped version"
-          echo "RELEASE_TAG=$(git describe --tags --abbrev=0)" >> $GITHUB_OUTPUT
-          git push --tags -f
-          git push
+          echo "previous-version=$(bump-my-version show current_version)" >> $GITHUB_OUTPUT
+          bump-my-version bump ${{ inputs.version_level }} --commit --tag
+          ([[ $? -gt 0 ]] && echo "bumped=false" || echo "bumped=true") >> $GITHUB_OUTPUT
+          echo "current-version=$(bump-my-version show current_version)" >> $GITHUB_OUTPUT
+      - name: Fail if bumping failes
+        if: steps.bump.outputs.bumped == 'false'
+        run: |
+          echo "Bumping failed."
+          git reset --hard HEAD^
+          exit 1
+      - name: Check new version number
+        if: steps.bump.outputs.bumped == 'true'
+        run: |
+          echo "Version was bumped from ${{ steps.bump.outputs.previous-version }} to ${{ steps.bump.outputs.current-version }}!"
 
+      - name: Merge main into develop
+        run: |
+          git checkout develop
+          git merge main --no-ff --no-edit
+      
+      - name: Push changes to GitHub
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: develop
+
+      - name: Checkout main
+        run: |
+          git checkout main
+
+      - name: Push changes to GitHub
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: main
+  
+  github_release:
+    name: Create a draft GitHub release
+    needs: merge_and_bump
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: main
       - name: Create GitHub Release
         id: create_release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          gh release create ${{ steps.bump.outputs.RELEASE_TAG }} \
-              --title="Release ${{ steps.bump.outputs.RELEASE_TAG }}" \
+          gh release create v${{ needs.merge_and_bump.outputs.new-version }} \
+              --title="Release v${{ needs.merge_and_bump.outputs.new-version }}" \
               --generate-notes \
               --draft
 
-      - name: Validate CITATION.cff
-        id: validate_cff
-        env:
-          GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
-        uses: dieghernan/cff-validator@v3
-
   tidy_workspace:
-    # only run if action above succeeds
+    name: Tidy up the repository
```

Comment:
This job now only removes deletes the branch from the repo. I think the name can reflect that better.